modules/maven/src/test/java/org/eclipse/sw360/antenna/maven/workflow/processors/MavenArtifactResolverTest.java

```diff
@@ -0,0 +1,136 @@
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.sw360.antenna.maven.workflow.processors;
+
+import org.apache.maven.plugin.LegacySupport;
+import org.apache.maven.project.MavenProject;
+import org.apache.maven.repository.RepositorySystem;
+import org.eclipse.sw360.antenna.api.IProject;
+import org.eclipse.sw360.antenna.api.configuration.ContextExtension;
+import org.eclipse.sw360.antenna.api.exceptions.AntennaException;
+import org.eclipse.sw360.antenna.maven.workflow.processors.enricher.MavenArtifactResolver;
+import org.eclipse.sw360.antenna.maven.workflow.processors.enricher.MavenArtifactResolverImpl;
+import org.eclipse.sw360.antenna.model.artifact.Artifact;
+import org.eclipse.sw360.antenna.model.artifact.facts.ArtifactMatchingMetadata;
+import org.eclipse.sw360.antenna.model.artifact.facts.java.MavenCoordinates;
+import org.eclipse.sw360.antenna.model.xml.generated.MatchState;
+import org.eclipse.sw360.antenna.testing.AntennaTestWithMockedContext;
+import org.eclipse.sw360.antenna.util.ProxySettings;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.nio.file.Path;
+import java.util.*;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.*;
+import static org.mockito.internal.verification.VerificationModeFactory.atLeast;
+
+public class MavenArtifactResolverTest extends AntennaTestWithMockedContext {
+    private IProject project;
+    private MavenArtifactResolver mavenArtifactResolver;
+    private List<Artifact> artifacts;
+
+    @Before
+    public void setUp() throws IOException {
+        System.setProperty("maven.home", "/usr/share/maven");
+        project = Mockito.mock(IProject.class);
+        when(antennaContextMock.getProject()).thenReturn(project);
+        org.apache.maven.repository.RepositorySystem mvnRepositorySystem = null;
+        when(antennaContextMock.getGeneric(org.apache.maven.repository.RepositorySystem.class))
+                .thenReturn(Optional.ofNullable(mvnRepositorySystem));
+        MavenProject mavenProject = new MavenProject();
+        when(antennaContextMock.getGeneric(MavenProject.class))
+                .thenReturn(Optional.of(mavenProject));
+        LegacySupport legacySupport = null;
+        when(antennaContextMock.getGeneric(LegacySupport.class))
+                .thenReturn(Optional.ofNullable(legacySupport));
+
+        when(configMock.getIgnoreForSourceResolving()).thenReturn(Collections.emptyList());
+        when(project.getBasedir()).thenReturn(new File(""));
+        when(toolConfigMock.getAntennaTargetDirectory()).thenReturn(temporaryFolder.newFolder("target").toPath());
```

Comment:
Better order of commands:
```suggestion
        Path targetDir = temporaryFolder.newFolder("target").toPath();
        when(toolConfigMock.getAntennaTargetDirectory()).thenReturn(targetDir);
```